File: src/Tests/Performance/PerfFib/PerfFib.go
package main

import (
	"fmt"
	"time"
)

// Recursive Fibonacci: measures raw function-call and integer-op overhead.

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	start := time.Now().UnixNano()

	sum := 0
	for i := 0; i < 5; i++ {
		sum += fib(34)
	}

	elapsed := time.Now().UnixNano() - start
	fmt.Println("checksum:", sum)
	fmt.Println("elapsed_ns:", elapsed)
}

File: src/Tests/Performance/PerfFib/PerfFib.cs
namespace go;

using fmt = fmt_package;
using time = time_package;

partial class main_package {

internal static nint fib(nint n) {
    if (n < 2) {
        return n;
    }
    return fib(n - 1) + fib(n - 2);
}

internal static void Main() {
    var start = time.Now().UnixNano();
    nint sum = 0;
    for (nint i = 0; i < 5; i++) {
        sum += fib(34);
    }
    var elapsed = time.Now().UnixNano() - start;
    fmt.Println("checksum:", sum);
    fmt.Println("elapsed_ns:", elapsed);
}

} // end main_package
